explainer.md

```diff
@@ -48,218 +49,298 @@ Provide web apps with efficient access to built-in (software and hardware) media
+  // Bind encode callbacks.
+  const reEncodeAudio = audioEncoder.encode.bind(audioEncoder);
+  const reEncodeVideo = videoEncode.encode.bind(videoEncoder);
 
-const audioEncoder = new AudioEncoder({
-  codec: 'opus',
-  settings: {
-    targetBitRate: 60_000,
-  },
-});
-const videoEncoder = new VideoEncoder({
-  codec: 'vp8',
-  settings: {
-    bitsPerSecond: 1_000_000,
-  },
-});
+  // Build decoders.
+  const audioDecoder = new AudioDecoder({ output: reEncodeAudio, error: onCodecError });
+  const videoDecoder = new VideoDecoder({ output: reEncodeVideo, error: onCodecError });
+
+  // Configure and reset if not supported. More sophisticated fallback recommended.
+  if (!await audioDecoder.configure({ codec: 'aac', ... }))
+    audioDecoder = null;
+  if (!await videoDecoder.configure({ codec : 'avc1.42001e', ... }))
+    videoDecoder = null;
+
+  return { audioDecoder: audioDecoder, videoDecoder: videoDecoder};
+}
+
+// Setup encoders.
+let { audioEncoder, videoEncoder } = await buildAndConfigureEncoders();
 
-const {audioIn, videoIn} = demux(input);
-const audioOut = audioIn.pipeThrough(audioDecoder).pipeThrough(audioEncoder);
-const videoOut = videoIn.pipeThrough(videoDecoder).pipeThrough(videoEncoder);
-const output = mux(audioOut, videoOut);
+// App handles unsupported configuration.
+if (audioEncoder == null || videoEncoder == null)
+  return;
 
+// Setup decoders. Provide encoders to receive decoded output.
+let { audioDecoder, videoDecoder } = await buildAndConfigureDecoders(audioEncoder, videoEncoder);
+
+// App handles unsupported configuration.
+if (audioDecoder == null || videoDecoder == null)
+  return;
+
+// Start streaming encoded chunks to the decoders, repeatedly calling
+// the provided callbacks for each chunk.
+// Decoded output will be fed to encoders for re-encoding.
+// Encoded output will be fed to muxer.
+streamEncodedChunks(audioDecoder.decode.bind(audioDecoder),
+          videoDecoder.decode.bind(videoDecoder));
 ```
 
 ### Example of real-time communication
 
 ```javascript
-// The app provides sources of audio and video, perhaps from getUserMedia.
+// The app provides sources of audio and video, perhaps from getUserMedia().
 const {audioTrack, videoTrack} = ...;
 // The app also provides ways to send encoded audio and video bitstream.
-function sendMedia(encodedAudio, encodedVideo) { ... };
+function sendAudio(encodedAudio) { ... };
+function sendVideo(encodedVideo) { ... };
 
-// First, the tracks are converted to ReadableStreams of framed bitstream chunks.
+// On the receive side, encoded media is likely received
+// from an out-of-order p2p transport and then put into a buffer.
+// The output of that buffer is the source of encoded audio and video here.
+function streamEncodedChunks(decodeAudioCallback, decodeVideoCallback) { ... }
+
+// App rendering audio in AudioWorklet
+// TODO(chcunningham): source a demo.
+function renderInAudioWorklet(audioPacket) { ... }
```

Comment:
@padenot roughly, I think this is something like the impl of [playSample() here](https://developer.mozilla.org/en-US/docs/Web/API/Web_Audio_API/Advanced_techniques), but connected to a worklet node for a worklet that just does bypass rendering. LMK if I'm on the right track.

Reply:
See more recent comment on l.63